### PR TITLE
GH-442 Refactor Widget testData call

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes GH-442

Signed-off-by: Gordon Smith <gordon.smith@lexisnexis.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@dtsnell4 Please Review 
@mlzummo, @jbrundage Can you take a look as well (as it touches every file).

There are some small changes I will be opening issues for, but this is a pretty important refactor.

**Comment by dtsnell4** [comment]
It all looks good to me.  There's still a few comments left in the "factory" js files though, if that is an issue.  And of course the Playground and other demos will need to be updated.

**Comment by GordonSmith** [comment]
@dtsnell4 Can you open issues for those?

**Comment by jbrundage** [comment]
This looks good to me

**Comment by mlzummo** [comment]
looks good to pull

**Comment by dtsnell4** [comment]
@GordonSmith should I open issues for the Playground, Theme Editor etc. to be updated to the new test data format?

**Comment by GordonSmith** [comment]
Yup I guess so.

File: test/Factory.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        d3.map(bundle.value).entries().forEach(function (widget) {
+            var widgetID = widget.key;
+            d3.map(widget.value).entries().forEach(function (sample) {
+                var sampleID = sample.key;
+                var widgetPath = "src/" + folderID + "/" + widgetID;
+                var widgetTestData = {
+                    folder: folderID,
+                    file: widgetID,
+                    sample: sampleID,
+                    widgetPath: widgetPath,
```
Comment (dtsnell4) [inline test/Factory.js]: @GordonSmith I have discovered what I believe is an issue with this line.  There is the possibility for some widgets, such as Composition, to not have a path (i.e. there is no "src/common/Composition").  This causes an issue when looping through all of of the widgets getting their paths, and I suspect would also when trying to create a Composition widget using createFunction() in Factory.js.

Comment (GordonSmith) [inline test/Factory.js]: I think the following variables are badly named:
```
var bundlesCatMap
var bundlesWidgetMap
```
And should really be called:
```
var folderName;
var fileName;
```

**Pull request merged**